lib/watir_drops/page_object.rb

```diff
@@ -18,52 +20,68 @@ def page_url
         end
       end
 
+      def page_title
+        define_method("page_title") do |args|
+          yield(*args)
+        end
+      end
+
       def element_list
         @element_list ||= []
       end
 
+      def required_element_list
+        @required_element_list ||= []
+      end
+
       def inherited(subclass)
         subclass.element_list = element_list.dup
+        subclass.required_element_list = required_element_list.dup
       end
 
-      def elements(name, &block)
+      def elements(name, required = false, &block)
         define_method(name) do |*args|
           self.instance_exec(*args, &block)
         end
 
         element_list << name.to_sym
+        required_element_list << name.to_sym if required
       end
 
-      def element(name, &block)
+      def element(name, required=false, &block)
         define_method(name) do |*args|
           self.instance_exec(*args, &block)
         end
 
         define_method("#{name}=") do |val|
           watir_element = self.instance_exec &block
           case watir_element
-          when Watir::Radio
-            watir_element.set if val
-          when Watir::CheckBox
-            val ? watir_element.set : watir_element.clear
-          when Watir::Select
-            watir_element.select val
-          when Watir::Button
-            watir_element.click
-          when Watir::TextField, Watir::TextArea
-            watir_element.set val if val
-          else
-            watir_element.click if val
+            when Watir::Radio
+              watir_element.set if val
+            when Watir::CheckBox
+              val ? watir_element.set : watir_element.clear
+            when Watir::Select
+              watir_element.select val
+            when Watir::Button
+              watir_element.click
+            when Watir::TextField, Watir::TextArea
+              watir_element.set val if val
+            else
+              watir_element.click if val
           end
         end
-
         element_list << name.to_sym
+        required_element_list << name.to_sym if required
       end
 
       def visit(*args)
-        new.tap { |page| page.goto(*args) }
+        new.tap do |page|
+          page.goto(*args)
+          fail "The page is not properly loaded!" unless page.on_page?
```

Comment:
required is added to ''page_rul" as well in latest push. since the exception will  be raised inside on_page? , can we simply use ```page.on_page?``` here?